components/api-resource-mgt/org.wso2.carbon.identity.api.resource.collection.mgt/src/main/java/org/wso2/carbon/identity/api/resource/collection/mgt/util/APIResourceCollectionMgtConfigBuilder.java

```diff
@@ -0,0 +1,179 @@
+            // Prevents using external document type definition when parsing xml.
+            factory.setProperty(XMLInputFactory.SUPPORT_DTD, false);
+            XMLStreamReader parser = factory.createXMLStreamReader(stream);
+            StAXOMBuilder builder = new StAXOMBuilder(parser);
+            documentElement = builder.getDocumentElement();
+            buildAPIResourceCollectionConfig();
+        } catch (IOException e) {
+            LOG.error("Error while loading API resource collection management configs.", e);
+        } catch (XMLStreamException e) {
+            LOG.error("Error while streaming API resource collection management configs.", e);
+        }
+    }
+
+    private void buildAPIResourceCollectionConfig() {
+
+        Iterator<OMElement> apiResourceCollections = this.documentElement.getChildrenWithName(
+                new QName(APIResourceCollectionConfigBuilderConstants.API_RESOURCE_COLLECTION_ELEMENT));
+
+        while (apiResourceCollections.hasNext()) {
+            OMElement apiResourceCollection = apiResourceCollections.next();
+            APIResourceCollection apiResourceCollectionObj =
+                    buildAPIResourceCollectionBasicInfo(apiResourceCollection);
+            if (apiResourceCollectionObj == null) {
+                continue;
+            }
+            // Fetch scopes.
+            OMElement scopesElement = apiResourceCollection.getFirstChildWithName(
+                    new QName(APIResourceCollectionConfigBuilderConstants.SCOPES_ELEMENT));
+            if (scopesElement != null) {
+                List<String> readScopeList = new ArrayList<>();
+                List<String> writeScopeList = new ArrayList<>();
+                Iterator<?> actionElements = scopesElement.getChildElements();
+                while (actionElements.hasNext()) {
+                    OMElement actionElement = (OMElement) actionElements.next();
+                    if (actionElement == null) {
+                        continue;
+                    }
+                    Iterator<OMElement> scopes = actionElement.getChildrenWithName(
+                            new QName(APIResourceCollectionConfigBuilderConstants.SCOPE_ELEMENT));
+                    while (scopes.hasNext()) {
+                        OMElement scope = scopes.next();
+                        String scopeName = scope.getAttributeValue(
+                                new QName(APIResourceCollectionConfigBuilderConstants.NAME));
+                        // Read, Feature scopes are considered as read scopes.
+                        if (APIResourceCollectionConfigBuilderConstants.READ.equals(actionElement.getLocalName()) ||
+                                APIResourceCollectionConfigBuilderConstants.FEATURE.equals(
+                                        actionElement.getLocalName())) {
+                            if (!readScopeList.contains(scopeName)) {
+                                readScopeList.add(scopeName);
```

Comment:
Shall we use `HashSet` for scope lists (`readScopeList`, `writeScopeList` ) to avoid duplicate checks.

Reply:
Fixed in https://github.com/wso2/carbon-identity-framework/pull/5807.